lib/models/room.js

```diff
@@ -10,6 +10,19 @@ var MatrixEvent = require("./event").MatrixEvent;
 var utils = require("../utils");
 var ContentRepo = require("../content-repo");
 
+function synthesizeReceipt(userId, event, receiptType) {
+    // This is really ugly because JS has no way to express an object literal
+    // where the name of a key comes from an expression
+    var fakeReceipt = {content: {}};
+    fakeReceipt.content[event.getId()] = {};
+    fakeReceipt.content[event.getId()][receiptType] = {};
+    fakeReceipt.content[event.getId()][receiptType][userId] = {
+        ts: event
+    };
+    return new MatrixEvent(fakeReceipt);
```

Comment:
Aha, you're right it doesn't have `user_id`. The spec has `room_id` and `type` though. My documentation was only documenting the `content` part since that's the part that is being manipulated.